src/jquery.interactive-world-map.js

```diff
@@ -1,142 +1,180 @@
+    }
 
-        });
-        //Configurations for the map. Dotted Style and popup prepended
-        colorHighlighted = $.fn.interactiveWorldMap.defaults.highlighted;
-        colorHover       = $.fn.interactiveWorldMap.defaults.hover;
-        pattern  = '<pattern id="dotsGeneral" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:#cccccc;"></circle></pattern>';
-        pattern2 = '<pattern id="dotsHighlighted" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:' + colorHighlighted + '"></circle></pattern>';
-        patternHover = '<pattern id="dotsHover" width="5" height="5" patternUnits="userSpaceOnUse"><circle x="1" y="1" cx="3" cy="3" r="3" style="stroke:none; fill:' + colorHover+ '"></circle></pattern>';
-        $(this).parent().prepend('<div class="main-modal"><div class="container"><div class="title"></div><div class="container-image" ><img src="" class="main-image" alt=""></div><p><h2></h2><div class="description"></div></p></div></div>');
-        $('defs').html(pattern+pattern2+patternHover);
-    };
+    function prependModalContainer(element) {
+        $(element).parent().prepend('<div class="main-modal"><div class="container"><div class="title"></div><div class="container-image" ><img src="" class="main-image" alt=""></div><p><h2></h2><div class="description"></div></p></div></div>');
+    }
 
+    function setOptions(element, options, defaults) {
+        $data = $(element).data();                                                      //Configuration coming from the data-* attributes
+        final = $.extend({}, defaults, $data);
 
-    $.fn.interactiveWorldMap.defaults = {
-        hover : "#FF0000",
-        highlighted: "#848484",
-        marginCountryPopupForTop: 100,
-        marginCountryPopupForLeft: 60
+        if(options) {
+            final = $.extend(final, options, options['colors'], options['margins']);   //Options coming from hash
+        }
+
+        marginCountryPopupForTop                                           = final['marginCountryPopupForTop'];
+        marginCountryPopupForLeft                                          = final['marginCountryPopupForTop'];
+        $.fn[pluginName].defaults.highlightedcoloroption                   = final['highlightedcoloroption'];
+        $.fn[pluginName].defaults.hovercoloroption                         = final['hovercoloroption'];
+        highlightColor                                                     = $.fn[pluginName].defaults.highlightedcoloroption;
+        hoverColor                                                         = $.fn[pluginName].defaults.hovercoloroption;
+        dataCountries                                                      = final['countries'];
     }
```

Comment:
As explained above, I think the whole init process can be much, much simpler. To give you a full picture, with the colors

```
// plugin
var defaults = {
  colors: {
    default: 'black',
    hover: 'red',
    highlight: 'blue'
  }
}

init: function() {
  this._settings = $.extend({}, this._defaults, this._options, $(this).data())
}

// JS call
$('#map').interactiveWorldMap({ colors: { hover: 'yellow' } });

// HTML
<div id="map" data-colors='{ "highlight" : "orange" }'>

// with the combination above this._settings results to
{
  colors: {
    default: 'black',
    hover: 'yellow',
    highlight: 'orange'
  }
}